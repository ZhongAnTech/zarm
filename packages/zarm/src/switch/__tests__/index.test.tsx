import React from 'react';
import { render, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Switch from '../index';

describe('Switch', () => {
  afterEach(() => {
    jest.restoreAllMocks();
  });

  describe('snapshot', () => {
    it('renders correctly', () => {
      const wrapper = render(<Switch />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('defaultChecked', () => {
      const wrapper = render(<Switch defaultChecked />);
      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });

  describe('behaviour', () => {
    it('should switch on if defaultChecked is true', () => {
      const wrapper = shallow(<Switch defaultChecked />);
      const inputWrapper = wrapper.find('input');
      expect(inputWrapper.prop('disabled')).toBeFalsy();
      expect(inputWrapper.prop('value')).toEqual('on');
      expect(inputWrapper.prop('defaultChecked')).toBeTruthy();
    });

    it('should switch off and disabled', () => {
      const wrapper = shallow(<Switch disabled />);
      const inputWrapper = wrapper.find('input');
      expect(inputWrapper.prop('disabled')).toBeTruthy();
      expect(inputWrapper.prop('value')).toEqual('off');
      expect(inputWrapper.prop('checked')).toBeFalsy();
    });

    it('should switch on and disabled', () => {
      const wrapper = shallow(<Switch defaultChecked disabled />);
      const inputWrapper = wrapper.find('input');
      expect(inputWrapper.prop('disabled')).toBeTruthy();
      expect(inputWrapper.prop('value')).toEqual('on');
      expect(inputWrapper.prop('defaultChecked')).toBeTruthy();
    });

    it('should handle change event without updating state', () => {
      const onChange = jest.fn();
      const wrapper = shallow(<Switch checked onChange={onChange} />);
      wrapper.find('input').simulate('change');
      expect(onChange).toBeCalledWith(false);
      expect(wrapper.find('input').prop('value')).toEqual('on');
    });

    it('should handle change event and update state if props.checked is not existed', () => {
      const onChange = jest.fn();
      const wrapper = shallow(<Switch onChange={onChange} />);
      expect(wrapper.find('input').prop('value')).toEqual('off');
      wrapper.find('input').simulate('change');
      expect(onChange).toBeCalledWith(true);
    });
  });
});
