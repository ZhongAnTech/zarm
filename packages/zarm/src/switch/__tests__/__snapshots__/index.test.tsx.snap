// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Switch snapshot defaultChecked 1`] = `
<span
  class="za-switch"
>
  <input
    checked=""
    class="za-switch__input"
    type="checkbox"
    value="on"
  />
</span>
`;

exports[`Switch snapshot renders correctly 1`] = `
<span
  class="za-switch"
>
  <input
    class="za-switch__input"
    type="checkbox"
    value="off"
  />
</span>
`;
